### PR TITLE
BI: sending commit metric

### LINKED ISSUE #29: Report events for lakefs-iceberg integration usage
We would like to report events for lakefs-iceberg integration in order to understand the actual usage of this integration.
We should think of a point (or more) in time of the user's usage that reflect the adoption of this integration.

@lynnro314 @johnnyaug

### PR DESCRIPTION
**Pull request by Jonathan-Rosenberg**
Once a table has changed (i.e. a new snapshot got created), a metric would be sent to lakeFS.

Closes #29 

## How was this tested?

Manually

FYI @NoyDavidson2

### CODE REVIEW

File: src/main/java/io/lakefs/iceberg/LakeFSReporter.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    public void logOp(String op) {
+        Map<String, Object> reportMap = new HashMap<>() {
+            {
+                put("class", "integration");
+                put("name", op);
+                put("count", 1);
+            }
+        };
+        try {
+            log(reportMap);
```
Comment (johnnyaug) [inline src/main/java/io/lakefs/iceberg/LakeFSReporter.java:42]: (probably messed up the indentation)

```suggestion
        try {
            log(Map.of("class", "integration",
                                "name", op,
                                "count", 1));
```

File: src/main/java/io/lakefs/iceberg/LakeFSReporter.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    private final String reportClient = "iceberg-catalog/" + getClass().getPackage().getImplementationVersion();
+
+    private LakeFSReporter(){}
+    public LakeFSReporter(Configuration hadoopConfig) {
+        String lakeFSServerURL = hadoopConfig.get("fs.s3a.endpoint");
+        if (lakeFSServerURL.endsWith("/")) {
+            lakeFSServerURL = lakeFSServerURL.substring(lakeFSServerURL.length()-1);
+        }
+        lakeFSServerURL += "/api/v1/statistics";
+        this.lakeFSEndpoint = URI.create(lakeFSServerURL);
```
Comment (johnnyaug) [inline src/main/java/io/lakefs/iceberg/LakeFSReporter.java:28]:
```suggestion
       this.lakeFSEndpoint = URI.create(StringUtils.stripEnd(lakeFSServerURL, "/") + "/api/v1/statistics");
```
(can also split to 2 lines)

**Review by johnnyaug — APPROVED** [review]
NICE!
Minor comments.

File: src/main/java/io/lakefs/iceberg/LakeFSReporter.java (hunk 2 above)
Comment (Jonathan-Rosenberg) [reply src/main/java/io/lakefs/iceberg/LakeFSReporter.java]: done

File: src/main/java/io/lakefs/iceberg/LakeFSReporter.java (hunk 1 above)
Comment (Jonathan-Rosenberg) [reply src/main/java/io/lakefs/iceberg/LakeFSReporter.java]: done

**Pull request merged**